include/rsl/algorithm.hpp

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <algorithm>
+#include <vector>
+
+namespace rsl {
+
+/**
+ * @brief Does the collection contain the value
+ * 
+ * @param collection   The input collection
+ * @param value        The value to test for in the collection
+ * 
+ * @return true if the value is found in the collection
+ */
+template<typename T>
+bool does_contain(std::vector<T> const& collection, T const& value) {
+  return std::find(collection.cbegin(), collection.cend(), value) != collection.cend();
+}
+
+/**
+ * @brief All the collection in the vector are unique
+ * 
+ * @param collection The input collection
+ * 
+ * @return true if all values in the collection are unique
+*/
+template <typename T>
+bool is_unique(std::vector<T> const& collection) {
+  auto copy_of_collection = collection;
+  std::sort(copy_of_collection.begin(), copy_of_collection.end());
+  return std::adjacent_find(copy_of_collection.cbegin(), copy_of_collection.cend()) == copy_of_collection.cend();
+}
```

Comment:
What's the purpose of making an internal copy instead of copying in the interface?

Reply:
That is a good point. I wanted to point out it makes a copy but it is easy enough to do that with a value parameter.